### PR TITLE
Entity list - dynamic columns

### PR DESCRIPTION
**Pull request by dback2**
- Change entity list columns to be data-driven
- Add ability for columns to be resized using the borders on the header cells
- Add multiselect columns dropdown in place of the previous "show extra info" toggle to allow for columns to be added and removed

Addresses:
https://highfidelity.manuscript.com/f/cases/1318/

Test Plan:

For the QA retest, you only need to quickly verify that resizing columns, adding/removing columns, and sorting columns continues to work as expected.

- Enter Interface in desktop mode into any domain with many different types of entities
- Open Create app to the entity list
- Adjust your search radius or your camera view (plus hitting refresh on the list) if you don't have a scrollable list with 50+ entities or so
- Verify that at the top of the scrollbar in place of the old right arrow to toggle showing extra columns there is now a down arrow dropdown that opens to a list of selectable columns as shown below
entitylistcolumndropdown
- Verify with the Columns dropdown open that when clicking anywhere outside of it (including when clicking to open the Types dropdown) that the Columns dropdown closes
- Verify the Types dropdown still performs as expected for adding/removing entity types
- Verify when opening the Types dropdown and that when clicking anywhere else outside of it (including when clicking to open the Columns dropdown) that the Types dropdown closes
- Verify when toggling the column checkboxes that the column is added or removed from the list as expected and that the other column widths adjust appropriately (they lose/gain width proportionally for columns added/removed based on their current percentage width of the table)
- Verify you can toggle sorting ascending/descending per column as expected and that the sort arrow shows at the top
- Verify when scrolling the list up and down that the column header row stays fixed at the top
- Verify when hovering over any of the borders between column headers that a resize cursor indicator appears
- Verify when selecting a column header border to resize that dragging it left lowers the left columns width by taking the width from the right column, and vice-versa, and that each row's cell for that column resizes with the header in parallel maintaining it's cell contents within the column header boundaries
- Verify you cannot drag the columns to width below the minimum which is as shown below
entitylistcolumnwidth
- Verify when resizing the entity list window that everything maintains their size proportions in the newly adjusted window and that it appears the same
- Verify when double clicking a cell in the name column to rename the entity that the black rename text field that appears is sized appropriately within the resized name cell and performs as expected
- Verify when lowering the radius or applying a search filter to reduce the entity list to not be scrollable that the columns adjust to fill the scrollbar space, and vice-versa
- Stress test adding/removing different columns, resizing the columns, sorting the columns, and resizing the window

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14885/HighFidelity-Beta-PR14369-0597d15.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14885/HighFidelity-Beta-PR14369-0597d15.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14885/HighFidelity-Beta-Interface-PR14369-0597d15.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14885/HighFidelity-Beta-PR14369-0597d15.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14885/HighFidelity-Beta-Interface-PR14369-0597d15.exe

**Comment by jakram** [comment]
Currently testing

File: scripts/system/html/js/entityList.js [hunk 1]
```diff
@@ -762,16 +851,109 @@ function loaded() {
...
+        function getColumnIndex(columnID) {
+            for (let i = 0; i < columns.length; ++i) {
+                if (columns[i].columnID === columnID) {
+                    return i;
+                }
+            }
+            return -1;
+        }
+        
+        function getColumnClassName(columnID) {
```
[the remark this answers is not in the document]
Comment (dback2) [reply scripts/system/html/js/entityList.js]: Agreed, makes sense

File: scripts/system/html/js/entityList.js [hunk 2]
```diff
@@ -205,52 +240,139 @@ function loaded() {
...
+            elResizer.setAttribute("columnIndex", columnIndex);
+            elResizer.onmousedown = onStartResize;
+            elTh.appendChild(elResizer);
+
+            let elSortOrder = document.createElement("span");
+            elSortOrder.className = "sort-order";
+            elTh.appendChild(elSortOrder);
+            elHeaderTr.appendChild(elTh);
+                        
+            elSortOrders[columnID] = document.querySelector('#' + thID + ' .sort-order');
```
[the remark this answers is not in the document]
Comment (dback2) [reply scripts/system/html/js/entityList.js]: Yep, nice catch

File: scripts/system/html/js/entityList.js [hunk 3]
```diff
@@ -205,52 +240,139 @@ function loaded() {
...
+            if (currentSortColumn === null) {
```
[the remark this answers is not in the document]
Comment (dback2) [reply scripts/system/html/js/entityList.js]: Done

**Comment by dback2** [comment]
For the QA retest, you only need to quickly verify that resizing columns, adding/removing columns, and sorting columns continues to work as expected.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15084/HighFidelity-Beta-PR14369-0b0614c.apk

**Comment by jakram** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15084/HighFidelity-Beta-PR14369-0b0614c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15084/HighFidelity-Beta-Interface-PR14369-0b0614c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15084/HighFidelity-Beta-PR14369-0b0614c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15084/HighFidelity-Beta-Interface-PR14369-0b0614c.exe

**Pull request merged**